simulator/src/main/java/simblock/node/consensus/ProofOfSpace.java

```diff
@@ -0,0 +1,68 @@
+package simblock.node.consensus;
+
+import simblock.block.Block;
+import simblock.block.ProofOfWorkBlock;
+import simblock.node.Node;
+import simblock.simulator.Simulator;
+import simblock.simulator.Timer;
+import simblock.task.AbstractMintingTask;
+import simblock.task.MiningTask;
+import simblock.task.WinningTask;
+
+import java.math.BigInteger;
+
+import static simblock.simulator.Main.random;
+
+public class ProofOfSpace extends AbstractConsensusAlgo{
+
+    public ProofOfSpace(){
```

Comment:
At the start of the simulation, we want to generate some number of events that signify when we check for a nodes turn. How would this work in a real network? Without a centralized authority, who checks for which nodes turn it is and determines if that node wins? How do other nodes verify that the winning node had a valid win - that it not only has a valid hash, but that the hash was submitted during the nodes turn?

Reply:
This should be refactored to more accurately represent a real network after understanding how to address the questions above

Reply:
What about storing the information about who is next as part of the new block? Say if it is my turn and my id is 10, then I insert a block and add to a field called turn an array of x (e.g., 3) next nodes. Probably in my case, it would be turn=[11, 12, 13] and now everybody knows that the next miner would be one of these. Therefore, if somebody propagates next node with creator id=14, people won't accept it. But what should we do if none of the 11, 12, 13 are alive? Obviously the next right candidate (says node 14) should step in and do the mining, but has to wait enough (mining period+propagation time in the network) to do so. This delay can be guaranteed through something like VDF (Verifiable Delay Functions) that guarantees that a node has waited enough before sending next block. 

For the sake of implementation, I think if we add a filed to nodes indicating the next miners (turn) will help. Do you agree?